src/Client/StoreOnChainWallet.php

```diff
@@ -0,0 +1,236 @@
+                    urlencode($storeId) . '/payment-methods' . '/OnChain' . '/' .
+                    urlencode($cryptoCode) . '/wallet' . '/feeRate';
+
+        if (isset($blockTarget)) {
+            $url .= '/?blockTarget=' . $blockTarget;
+        }
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\StoreOnChainWalletFeeRate(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getStoreOnChainWalletAddress(
+        string $storeId,
+        string $cryptoCode,
+        ?string $forceGenerate = 'false'
+    ): \BTCPayServer\Result\StoreOnChainWalletAddress {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/payment-methods' . '/OnChain' . '/' .
+                    urlencode($cryptoCode) . '/wallet' . '/address';
+
+        if (isset($forceGenerate)) {
+            $url .= '/?forceGenerate=' . $forceGenerate;
+        }
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\StoreOnChainWalletAddress(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function unReserveLastStoreOnChainWalletAddress(
+        string $storeId,
+        string $cryptoCode
+    ): bool {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/payment-methods' . '/OnChain' . '/' .
+                    urlencode($cryptoCode) . '/wallet' . '/address';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'DELETE';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return true;
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getStoreOnChainWalletTransactions(
+        string $storeId,
+        string $cryptoCode,
+        ?array $statusFilters = null,
+        ?int $skip = null,
+        ?int $limit = null
+    ): \BTCPayServer\Result\StoreOnChainWalletTransactionList {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/payment-methods' . '/OnChain' . '/' .
+                    urlencode($cryptoCode) . '/wallet' . '/transactions/?';
+
+        //add each status filter to the query if it is set
```

Comment:
Nitpick for consistency, we should keep the comments same, I prefer to start uppercase and end with a . like:
```
// Add each status filter to the query if it is set.
```